engine/gamesys/src/gamesys/scripts/script_box2d_body.cpp

```diff
@@ -0,0 +1,885 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include <stdio.h>
+
+#include <Box2D/Dynamics/b2Body.h>
+#include <Box2D/Dynamics/Joints/b2Joint.h>
+
+#include <dlib/log.h>
+#include <gameobject/script.h>
+
+#include "gamesys.h"
+#include "../gamesys_private.h"
+
+#include "script_box2d.h"
+
+extern "C"
+{
+#include <lua/lauxlib.h>
+#include <lua/lualib.h>
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// b2Body
+namespace dmGameSystem
+{
+
+    static uint32_t TYPE_HASH_BODY = 0;
+
+    #define BOX2D_TYPE_NAME_BODY "b2body"
+
+    void PushBody(lua_State* L, b2Body* body)
+    {
+        b2Body** bodyp = (b2Body**)lua_newuserdata(L, sizeof(b2Body*));
+        *bodyp = body;
+        luaL_getmetatable(L, BOX2D_TYPE_NAME_BODY);
+        lua_setmetatable(L, -2);
+    }
+
+    b2Body* CheckBody(lua_State* L, int index)
+    {
+        b2Body** pbody = (b2Body**)dmScript::CheckUserType(L, index, TYPE_HASH_BODY, "Expected user type " BOX2D_TYPE_NAME_BODY);
+        return *pbody;
+    }
+
+    static b2Body* ToBody(lua_State* L, int index)
+    {
+        return (b2Body*)dmScript::ToUserType(L, index, TYPE_HASH_BODY);
+    }
+
+    static int Body_GetPosition(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        b2Body* body = CheckBody(L, 1);
+        dmScript::PushVector3(L, FromB2(body->GetPosition(), GetInvPhysicsScale()));
```

Comment:
Just like in our old implementation, we need to take take of conversion between units, using FromB2/CheckVec2.